src/error.rs

```diff
@@ -1326,17 +1326,30 @@ impl ToDiagnostic<FileId> for TypecheckError {
                         format!("Maybe you forgot to put a `forall {}.` somewhere in the enclosing type ?", ident),
                     ])]
             ,
-            TypecheckError::TypeMismatch(expd, actual, span_opt) =>
+            TypecheckError::TypeMismatch(expd, actual, span_opt) => {
+                fn addendum(ty: &Types) -> String {
```

Comment:
Nitpick: I think you can use `&str` instead of `String` here if you just need to print them, avoiding an allocation.